### PR TITLE
[TypeCheck] Add 'Renamed' and clean things up

### PR DESCRIPTION
**Pull request by effectfully**
Just a small refactoring. `Renamed` is something that I'm going to advertise to the QuviQ people. Even if they don't end up using it, it's just one `newtype` definition and a single accompanying one-liner, so given the utility I believe it's worth having it in the codebase.

### CODE REVIEW

File: plutus-core/plutus-core/src/PlutusCore/Rename.hs [hunk 1]
```diff
@@ -60,11 +66,17 @@ instance Rename a => Rename (Normalized a) where
 -- 2. some value may be used multiple times
 --
 -- so we annotate such a value with 'Dupable' and call 'liftDupable' at each usage, which ensures
--- global conditions is preserved.
+-- global uniqueness is preserved.
+--
+-- 'unDupable' is not supposed to be exported. Don't provide any instances allowing the user to
+-- access the underlying value.
 newtype Dupable a = Dupable
```
Comment (michaelpj) [inline plutus-core/plutus-core/src/PlutusCore/Rename.hs:73]: I wonder if you want to define this next to `Renamed` and discuss them together. They have sort of modal logic relationship:
- `Renamed` == necessarily globally unique
- `Dupable` == not necessarily globally unique == possibly not globally unique

Maybe they could have more symmetrical names too. I don't have great suggestions ("Renamed" is a bit about what's *happened* to it not the *property*). "GloballyUnique"/"NonGloballyUnique"?

**Review by michaelpj — APPROVED** [review]
Idea seems good to me. I wonder if there are places where it would be sensible for us to use it?

Actually, there's definitely a place we should use `Dupable` that we don't: the inliner!

**Comment by michaelpj** [comment]
I made https://input-output.atlassian.net/browse/PLT-339 for the inliner issue.

**Comment by effectfully** [comment]
Hydra seems to be stuck.

**Pull request merged**